### ISSUE TITLE
POM breaks with org.eclipse.xtext.xbase.lib

### ISSUE THREAD

**Issue by fnmartinez** [sarl/sarl#327]
Hi, I was using SARL along with Spring-Boot to use Spring-MVC to show the agents state, and it happens to break with the following message:

> Couldn't find a valid version of the Xtext base library in the classpath. You should add 'org.eclipse.xtext.xbase.lib', version 2.8.1 or higher.

This only happens on the SARL IDE, Maven runs and builds the program well.

The POM file is the following:

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>

	<groupId>ar.edu.itba.it.sma</groupId>
	<artifactId>rigged</artifactId>
	<version>1.0-SNAPSHOT</version>
	<packaging>jar</packaging>

	<name>bote</name>
	<url>http://maven.apache.org</url>

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<java.version>1.7</java.version>
		<sarl.version>0.2.0</sarl.version>
		<sarl.plugin.version>0.2.0</sarl.plugin.version>
		<janus.version>2.0.2.0</janus.version>
	</properties>

	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>1.2.6.RELEASE</version>
	</parent>

	<dependencies>
		<dependency>
			<groupId>io.janusproject</groupId>
			<artifactId>io.janusproject.kernel</artifactId>
			<version>${janus.version}</version>
		</dependency>
<!-- 		<dependency>
			<groupId>io.sarl.maven</groupId>
			<artifactId>io.sarl.maven.sdk</artifactId>
			<version>${sarl.version}</version>
			<type>pom</type>
			<exclusions>
				<exclusion>
					<groupId>com.google.guava</groupId>
					<artifactId>guava</artifactId>
				</exclusion>
			</exclusions>
		</dependency> -->
		<dependency>
			<groupId>org.eclipse.xtext</groupId>
			<artifactId>org.eclipse.xtext.xbase.lib</artifactId>
			<version>2.8.4</version>
		</dependency>
		<dependency>
			<groupId>com.vaadin</groupId>
			<artifactId>vaadin-spring-boot-starter</artifactId>
			<version>1.0.0</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-websocket</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>

	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.vaadin</groupId>
				<artifactId>vaadin-bom</artifactId>
				<version>7.5.5</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>

	<build>
		<resources>
			<resource>
				<directory>src/main/sarl</directory>
				<includes>
					<include>**/*.sarl</include>
				</includes>
			</resource>
		</resources>
		<plugins>
			<plugin>
				<groupId>io.sarl.maven</groupId>
				<artifactId>sarl-maven-plugin</artifactId>
				<version>${sarl.plugin.version}</version>
				<extensions>true</extensions>
				<configuration>
					<source>${java.version}</source>
					<target>${java.version}</target>
				</configuration>
			</plugin>
			<plugin>
				<!-- to create a jar with all dependencies -->
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-assembly-plugin</artifactId>
				<version>2.5.3</version>
				<configuration>
					<descriptors>
						<descriptor>with-dependencies.xml</descriptor>
					</descriptors>
					<archive>
						<manifest>
							<mainClass>io.janusproject.Boot</mainClass>
						</manifest>
					</archive>
				</configuration>
				<executions>
					<execution>
						<id>make-assembly</id>
						<phase>package</phase>
						<goals>
							<goal>attached</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by gallandarakhneorg** [comment]
First, I recommend you to use the version `1.0.0-SNAPHOT` of SARL and the version `2.1.0.0-SNAPSHOT` of the Janus runtime environment. Even if they are still flagged as unstable, they contains less bugs than the version 0.2.0. The maven servers to use are `maven.sarl.io` and `maven.janusproject.io`.

Could you check in the "About Eclipse" the versions of SARL and Xtext and give us a feedback?
During the past, this error occurs when the Eclipse product has a version different than the one in the Maven dependencies.

**Comment by gallandarakhneorg** [comment]
You could also remove the dependency to `org.eclipse.xtext.xbase.lib` in your POM, since this module is included by the SARL/Janus dependency.

**Comment by fnmartinez** [comment]
Hi! Thank you for your quick reply. I changed the versions to Janus 2.1.0.0-SNAPSHOT in your repos, and it worked like magic. Thank you very much!

Here I attach a screenshot with the version for XText. 2.8.1, it seems.
![about-sarl-development-environment](https://cloud.githubusercontent.com/assets/774253/10727033/892c4ac4-7bb5-11e5-88f1-d3e64c6265de.png)

Final POM follows:
```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>

	<groupId>ar.edu.itba.it.sma</groupId>
	<artifactId>rigged</artifactId>
	<version>1.0-SNAPSHOT</version>
	<packaging>jar</packaging>

	<name>bote</name>
	<url>http://maven.apache.org</url>

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<java.version>1.7</java.version>
		<sarl.plugin.version>0.2.0</sarl.plugin.version>
		<janus.version>2.1.0.0-SNAPSHOT</janus.version>
	</properties>

	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>1.2.6.RELEASE</version>
	</parent>

	<dependencies>
		<dependency>
			<groupId>io.janusproject</groupId>
			<artifactId>io.janusproject.kernel</artifactId>
			<version>${janus.version}</version>
		</dependency>
		<dependency>
			<groupId>com.vaadin</groupId>
			<artifactId>vaadin-spring-boot-starter</artifactId>
			<version>1.0.0</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-websocket</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>

	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.vaadin</groupId>
				<artifactId>vaadin-bom</artifactId>
				<version>7.5.5</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>

	<build>
		<resources>
			<resource>
				<directory>src/main/sarl</directory>
				<includes>
					<include>**/*.sarl</include>
				</includes>
			</resource>
		</resources>
		<plugins>
			<plugin>
				<groupId>io.sarl.maven</groupId>
				<artifactId>sarl-maven-plugin</artifactId>
				<version>${sarl.plugin.version}</version>
				<extensions>true</extensions>
				<configuration>
					<source>${java.version}</source>
					<target>${java.version}</target>
				</configuration>
			</plugin>
			<plugin>
				<!-- to create a jar with all dependencies -->
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-assembly-plugin</artifactId>
				<version>2.5.3</version>
				<configuration>
					<descriptors>
						<descriptor>with-dependencies.xml</descriptor>
					</descriptors>
					<archive>
						<manifest>
							<mainClass>io.janusproject.Boot</mainClass>
						</manifest>
					</archive>
				</configuration>
				<executions>
					<execution>
						<id>make-assembly</id>
						<phase>package</phase>
						<goals>
							<goal>attached</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>
	
	<repositories>
		<repository>
			<id>janusproject</id>
			<name>Janus Project Repo</name>
			<url>http://maven.janusproject.io/</url>
		</repository>
	</repositories>
</project>
```

**Comment by gallandarakhneorg** [comment]
Thank you for your feedback.
We will check the version of Xtext in the SARL product.

**Comment by gallandarakhneorg** [comment]
In your POM, you should use a version of SARL that iss compatible with the one of Janus.
So that:
```
<sarl.plugin.version>1.0.0-SNAPSHOT</sarl.plugin.version>
<janus.version>2.1.0.0-SNAPSHOT</janus.version>
```

According to the page of [Janus](https://www.janusproject.io/#versionnumber), if the version of Janus is W.X.Y.Z-SNAPSHOT, the nthe version of SARL should be X.Y.A-SNAPSHOT. Otherwise, you may encounter compatibility problems between the API used in the code generated by the SARL compiler and the API used by the runtime environment.

**Comment by gallandarakhneorg** [comment]
This issue could be closed since the issue #302 has the goal to add version incompatibility detector.
